src/qudi/core/modulemanager.py

```diff
@@ -246,6 +248,74 @@ def _qudi_main_ref_dead_callback(self):
                      'ModuleManager.')
         self.clear()
 
+    def dump_status_variables(self):
+        """
+        Method that dumps the status variables of all active modules.
+        """
+        logger.debug(f"Dumping status variables")
+        with self._lock:
+            for _, module in self.modules.items():
+                if module.is_active:
+                    module.instance._dump_status_variables()
+
+    def toggle_automated_status_variable_dumping(self, toggle):
+        """
+        Method that creates or destroys the QTimer for handling the automatic dumping of status variables.
+
+        @param bool toggle: boolean that determines whether timer is created or destroyed.
+        """
+        if not toggle:
+            logger.info(f"Automated status variable saving disabled.")
+            self.automated_status_variable_dumping_timer.stop()
+            try:
+                self.automated_status_variable_dumping_timer.timeout.disconnect()
+            except RuntimeError:
+                pass
+            return
+
+        logger.info(
+            f"Automated status variable saving enabled with interval {self.automated_status_variable_dumping_timer_interval} min."
+        )
+
+        self.automated_status_variable_dumping_timer.setInterval(
+            self.automated_status_variable_dumping_timer_interval * 60e3
+        )
+        self.automated_status_variable_dumping_timer.timeout.connect(
+            self.dump_status_variables, QtCore.Qt.QueuedConnection
+        )
+        self.automated_status_variable_dumping_timer.start()
+
+    @property
+    def automated_status_variable_dumping_timer_interval(self):
+        """
+        Property for the timer interval of the automatic status variable saving in min.
+
+        @return int: timer interval in min
+        """
+        return self._automated_status_variable_dumping_timer_interval
+
+    @automated_status_variable_dumping_timer_interval.setter
+    def automated_status_variable_dumping_timer_interval(self, interval):
```

Comment:
There should be some value checking. From non-gui, interval=0 will probably cause issues.

Reply:
I introduced an Exception that is thrown if the interval is <= 0.